src/core/ext/filters/client_channel/subchannel_index.c

```diff
@@ -119,15 +121,22 @@ static const gpr_avl_vtable subchannel_avl_vtable = {
 void grpc_subchannel_index_init(void) {
   g_subchannel_index = gpr_avl_create(&subchannel_avl_vtable);
   gpr_mu_init(&g_mu);
+  gpr_ref_init(&g_refcount, 1);
 }
 
-void grpc_subchannel_index_shutdown(void) {
-  grpc_exec_ctx exec_ctx = GRPC_EXEC_CTX_INIT;
-  gpr_mu_destroy(&g_mu);
-  gpr_avl_unref(g_subchannel_index, &exec_ctx);
-  grpc_exec_ctx_finish(&exec_ctx);
+void grpc_subchannel_index_shutdown(void) { grpc_subchannel_index_unref(); }
```

Comment:
Please add a TODO here indicating that we still need to do something to force polling in order to flush any pending callbacks.

Reply:
Sure. I should have done this.